application/backend/src/core/logging.py

```diff
@@ -0,0 +1,198 @@
+    """Configure application-wide logging with worker-specific log files.
+
+    Creates separate log files for each worker type (training, inference, etc.) with
+    configurable rotation and retention. Logs are filtered by the 'worker' field in
+    record extras and serialized as JSON.
+
+    Args:
+        config: Optional LogConfig instance. If None, uses default configuration.
+
+    Important: Must be called in each child process (BaseProcessWorker does this
+    automatically) and once at main process startup. Loguru sinks don't transfer
+    to child processes.
+
+    Example:
+        >>> setup_logging()
+        # Creates: logs/training.log, logs/inference.log, logs/app.log, etc.
+        >>> custom_config = LogConfig(rotation="50 MB", level="INFO")
+        >>> setup_logging(custom_config)
+    """
+    if config is None:
+        config = LogConfig()
+
+    log_config = config
+
+    for worker_name, log_file in log_config.worker_log_info.items():
+
+        def worker_log_filter(record: "Record", worker: str | None = worker_name) -> bool:
+            return record["extra"].get("worker") == worker
+
+        log_path = os.path.join(config.log_folder, log_file)
+
+        try:
+            os.makedirs(os.path.dirname(log_path), exist_ok=True)
+        except OSError as e:
+            logger.warning(f"Failed to create log directory {log_path}: {e}")
+            continue
+
+        try:
+            logger.add(
+                log_path,
+                rotation=config.rotation,
+                retention=config.retention,
+                level=config.level,
+                filter=worker_log_filter,
+                serialize=config.serialize,
+                enqueue=True,
+            )
+        except Exception as e:
+            logger.error(f"Failed to add log sink for {worker_name}: {e}")
+
+
+def get_job_logs_path(job_id: str | UUID) -> str:
+    """Get the path to the logs folder for a specific job.
+
+    Args:
+        job_id: Unique identifier for the job
+
+    Returns:
+        str: Path to the job's logs folder (logs/jobs/{job_id})
+
+    Raises:
+        ValueError: If job_id contains invalid characters
+
+    Example:
+        >>> get_job_logs_path(job_id="foo-123")
+        'logs/jobs/foo-123'
+    """
+    job_id = _validate_job_id(job_id)
+    jobs_folder = os.path.join(log_config.log_folder, "jobs")
+    try:
+        os.makedirs(jobs_folder, exist_ok=True)
+    except OSError as e:
+        raise RuntimeError(f"Failed to create jobs log directory: {e}") from e
+    return os.path.join(jobs_folder, f"{job_id}.log")
+
+
+@contextmanager
+def job_logging_ctx(job_id: str | UUID) -> Generator[str, None, None]:
+    """Add a temporary log sink for a specific job.
+
+    Captures all logs emitted during the context to logs/jobs/{job_id}.log.
+    The sink is automatically removed on exit, but the log file persists.
+    Logs also continue to go to other configured sinks.
```

Comment:
Question/clarification: based on what I've read this won't be returned by the project's sinks endpoints right? So it's not a pipeline sink?